### ISSUE TITLE
Openings should be flexboxes

### ISSUE THREAD

**Issue by dglazkov** [Polymer/core-scaffold#10]
I just ran into a situation where I really needed the content opening of `core-scaffold` to be a flexbox, so that I could size the content easily. It seems that if we're relying on flexbox to be the main layout mechanism, we should just make all openings flexey (probably `flex-direction:column`, to be more usual block styling. WDYT? It takes one extra div to get out of the flexbox if you don't want it, but it's super-hacky (`core-scaffold::shadow ::shadow #mainContainer`) to do the other way around.

**Comment by MelHarbour** [comment]
@robdodson Came up against a similar thought when trying to implement a material design search box using core-scaffold. The design spec says that we should make the menu button hidden, and replaced by the back icon, but there's no way I can see to cleanly access the menu button without breaking encapsulation. I think this tends to suggest that core-scaffold is a good general purpose starter example, but with the heavy caveat that if you want to do anything 'interesting', you're going to have to reimplement it youself, as @frankiefu suggests.

**Comment by robdodson** [comment]
yeah, for the [yeoman generator](https://github.com/yeoman/generator-polymer) I've taken to just using core-drawer-panel as the scaffold.
